app/workers/historical_data_deletion_worker.rb

```diff
@@ -42,4 +42,10 @@ def empty_subscriber_lists
                   .where("subscriber_lists.created_at < ?", empty_list_retention_period)
                   .where("subscriptions.id": nil)
   end
+
+  def historic_nullified_subscribers
+    Subscriber.nullified
```

Comment:
I've gone with your approach of creating both within a transaction.

Reply:
Ah, so by "the above" I meant "the previous/first" 😬. Sorry, I should have been more explicit. It still seems reasonable to create the subscriber in a transaction 👍. The reason the first approach is safer is because we can't predict what future code may come along, which might create subscribers without a subscription. In general, it's a bit odd to be deleting subscribers soon after they've been created, so waiting a year (in place of "nullified").

Reply:
Ah okay I get where you're coming from.

We can't trust the system... 😢

Well I've added the filter so the historic worker will only delete subscribers over a year old.

Reply:
Thanks! Looks good to me 👍.